trace_decoder/src/compact/compact_to_partial_trie.rs

```diff
@@ -8,183 +11,261 @@ use mpt_trie::{
         }
     }
+}
 
-    Ok(())
+#[derive(Debug, Default)]
+pub(super) struct StateTrieExtractionOutput {
+    pub(super) trie: HashedPartialTrie,
+    pub(super) code: HashMap<CodeHash, Vec<u8>>,
+    pub(super) storage_tries: HashMap<HashedAccountAddr, HashedPartialTrie>,
 }
 
-fn process_code(
-    c_bytes: Vec<u8>,
-    output: &mut CompactToPartialTrieExtractionOutput,
-) -> CompactParsingResult<()> {
-    let c_hash = hash(&c_bytes);
-    output.code.insert(c_hash, c_bytes);
+impl CompactToPartialTrieExtractionOutput for StateTrieExtractionOutput {
+    fn process_code(&mut self, c_bytes: Vec<u8>) -> CompactParsingResult<()> {
+        let c_hash = hash(&c_bytes);
+        self.code.insert(c_hash, c_bytes);
 
-    Ok(())
-}
+        Ok(())
+    }
 
-fn process_empty() -> CompactParsingResult<()> {
-    // Nothing to do.
-    Ok(())
+    fn process_leaf(
+        &mut self,
+        curr_key: Nibbles,
+        leaf_key: &Nibbles,
+        leaf_node_data: &LeafNodeData,
+    ) -> CompactParsingResult<()> {
+        process_leaf_common(
+            &mut self.trie,
+            curr_key,
+            leaf_key,
+            leaf_node_data,
+            |acc_data, full_k| {
+                Ok(
+                    convert_account_node_data_to_rlp_bytes_and_add_any_code_to_lookup(
+                        acc_data,
+                        full_k,
+                        &mut self.code,
+                        &mut self.storage_tries,
+                    ),
+                )
+            },
+        )
+    }
+    fn get_trie(&mut self) -> &mut HashedPartialTrie {
+        &mut self.trie
+    }
+}
+#[derive(Debug, Default)]
+pub(super) struct StorageTrieExtractionOutput {
+    pub(super) trie: HashedPartialTrie,
 }
 
-fn process_hash(
-    curr_key: Nibbles,
-    hash: TrieRootHash,
-    p_trie: &mut HashedPartialTrie,
-) -> CompactParsingResult<()> {
-    // If we see a hash node at this stage, it must be a hashed out node in the
-    // trie.
-    p_trie.insert(curr_key, hash);
+impl CompactToPartialTrieExtractionOutput for StorageTrieExtractionOutput {
+    fn process_code(&mut self, c_bytes: Vec<u8>) -> CompactParsingResult<()> {
+        Err(CompactParsingError::UnexpectedNodeForTrieType(
+            UnexpectedCompactNodeType::Code,
+            TrieType::Storage,
+        ))
+    }
 
-    Ok(())
+    fn process_leaf(
+        &mut self,
+        curr_key: Nibbles,
+        leaf_key: &Nibbles,
+        leaf_node_data: &LeafNodeData,
+    ) -> CompactParsingResult<()> {
+        process_leaf_common(
+            &mut self.trie,
+            curr_key,
+            leaf_key,
+            leaf_node_data,
+            |_, _| {
+                Err(CompactParsingError::UnexpectedNodeForTrieType(
+                    UnexpectedCompactNodeType::AccountLeaf,
+                    TrieType::Storage,
+                ))
+            },
+        )
+    }
+
+    fn get_trie(&mut self) -> &mut HashedPartialTrie {
```

Comment:
In idiomatic Rust, `get_` is frowned upon (https://rust-lang.github.io/api-guidelines/naming.html#c-getter) so this should be named just `trie()`.

Reply:
We have 131 getter methods named `get_foo` in the overall monorepo though 😬

Reply:
🤷 We can't fix all the wrongs in the world in one PR, but that's not a reason to keep doing it wrong?

Reply:
Hmm... So I would say this isn't so much a `getter`/`setter` as it is a way to get a common field from a concrete type implementing a trait. I'm actually very aligned with the community here in that I really don't like using `getter`s & `setter`s. There's no other way to get a common field from a trait however, so I think this is needed. We don't need to prefix it with `get_` however, so we could change that.